cirq-core/cirq/interop/quirk/cells/arithmetic_cells.py

```diff
@@ -148,6 +150,109 @@ def __repr__(self) -> str:
         )
 
 
+@value.value_equality
+class QuirkArithmeticGate(ops.ArithmeticGate):
+    """Applies arithmetic to a target and some inputs.
+
+    Implements Quirk-specific implicit effects like assuming that the presence
+    of an 'r' input implies modular arithmetic.
+
+    In Quirk, modular operations have no effect on values larger than the
+    modulus. This convention is used because unitarity forces *some* convention
+    on out-of-range values (they cannot simply disappear or raise exceptions),
+    and the simplest is to do nothing. This call handles ensuring that happens,
+    and ensuring the new target register value is normalized modulo the modulus.
+    """
+
+    def __init__(
+        self, identifier: str, target: Sequence[int], inputs: Sequence[Union[Sequence[int], int]]
+    ):
+        """Inits QuirkArithmeticGate.
+
+        Args:
+            identifier: The quirk identifier string for this operation.
+            target: The target qubit register.
+            inputs: Qubit registers (or classical constants) that
+                determine what happens to the target.
```

Comment:
It would be good to clarify that a "qubit register" means a sequence of qid shapes